bootstore/src/schemes/v0/state_initial_member.rs

```diff
@@ -0,0 +1,411 @@
+    ) -> (State, Output) {
+        use ResponseType::*;
+        let output = match response {
+            InitAck => {
+                if let Some(rack_init_state) = &mut self.rack_init_state {
+                    if rack_init_state.on_ack(from) {
+                        return (
+                            self.into(),
+                            ApiOutput::RackInitComplete.into(),
+                        );
+                    }
+                }
+                Output::none()
+            }
+            Share(share) => self.on_share(common, from, request_id, share),
+            Pkg(_) => ApiError::UnexpectedResponse {
+                from,
+                state: self.name(),
+                request_id,
+                msg: response.name(),
+            }
+            .into(),
+            Error(error) => ApiError::ErrorResponseReceived {
+                from,
+                state: self.name(),
+                request_id,
+                error,
+            }
+            .into(),
+        };
+
+        // This is a terminal state
+        (self.into(), output)
+    }
+
+    fn tick(mut self, common: &mut FsmCommonData) -> (State, Output) {
+        // Check for rack initialization timeout
+        if let Some(rack_init_state) = &mut self.rack_init_state {
+            if rack_init_state
+                .timer_expired(common.clock, common.config.rack_init_timeout)
+            {
+                let unacked_peers = rack_init_state.unacked_peers();
+                *rack_init_state = RackInitState::Timeout;
+                return (
+                    self.into(),
+                    ApiError::RackInitTimeout { unacked_peers }.into(),
+                );
+            }
+        }
+        self.pending_learn_requests
+            .retain(|_, metadata| metadata.expiry >= common.clock);
+
+        (self.into(), common.rack_secret_state.on_tick(common.clock))
+    }
+
+    fn on_connect(
+        &mut self,
+        common: &mut FsmCommonData,
+        peer: Baseboard,
+    ) -> Output {
+        common.on_connect(peer, self.pkg.rack_uuid)
+    }
+
+    fn on_disconnect(
+        &mut self,
+        _common: &mut FsmCommonData,
+        _peer: Baseboard,
+    ) -> Output {
+        // TODO: Discard any learn requests from this peer?
+        // The will expire anyway on timeout, but they may eat a share if the
+        //timeout doesn't occur
+        Output::none()
+    }
+}
+
+// Send a `ResponseType::Share` message once we have recomputed the rack secret
+fn decrypt_and_send_share_response(
+    from: Baseboard,
+    request_id: Uuid,
+    pkg: &SharePkgV0,
+    distributed_shares: &mut BTreeMap<Baseboard, ShareIdx>,
+    rack_secret: &RackSecret,
+) -> Output {
+    match pkg.decrypt_shares(rack_secret) {
+        Ok(shares) => send_share_response(
+            from,
+            request_id,
+            pkg,
+            distributed_shares,
+            &shares,
+        ),
+        Err(_) => Output::respond(
```

Comment:
This is Real Bad, right?  Maybe add a comment to that effect here.

Reply:
Added a comment in [037aaa2](https://github.com/oxidecomputer/omicron/pull/3385/commits/037aaa23dd141bf9e107624dc25ae1f99215dd3f)